lib/api/objectPutRetention.js

```diff
@@ -0,0 +1,116 @@
+const async = require('async');
+const { errors, s3middleware } = require('arsenal');
+
+const { decodeVersionId, getVersionIdResHeader } =
+  require('./apiUtils/object/versioning');
+
+const { metadataValidateBucketAndObj } = require('../metadata/metadataUtils');
+const { pushMetric } = require('../utapi/utilities');
+const getReplicationInfo = require('./apiUtils/object/getReplicationInfo');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const metadata = require('../metadata/wrapper');
+
+const { parseRetentionXml } = s3middleware.retention;
+const REPLICATION_ACTION = 'PUT_RETENTION';
+
+/**
+ * Object Put Retention - Adds retention information to object
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+function objectPutRetention(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'objectPutRetention' });
+
+    const { bucketName, objectKey } = request;
+
+    const decodedVidResult = decodeVersionId(request.query);
+    if (decodedVidResult instanceof Error) {
+        log.trace('invalid versionId query', {
+            versionId: request.query.versionId,
+            error: decodedVidResult,
+        });
+        return process.nextTick(() => callback(decodedVidResult));
+    }
+    const reqVersionId = decodedVidResult;
```

Comment:
Not sure creating a second variable here is helpful, could of just kept using the original, maybe a new name but two copies seems a bit much